src/ProtoValidate/Internal/Evaluator/ConstraintResolver.cs

```diff
@@ -79,7 +74,7 @@ public FieldConstraints ResolveFieldConstraints(FieldDescriptor descriptor)
         var options = descriptor.GetOptions();
         if (options == null || !options.HasExtension(ValidateExtensions.Field))
         {
-            return new FieldConstraints();
+            return null;
```

Comment:
This results in a compiler warning.  The return type should be changed to FieldConstraints? to avoid the warning.